src/LanguageServer.ts

```diff
@@ -816,11 +801,14 @@ export class LanguageServer {
         } else if (program.hasFile(change.pathAbsolute)) {
             //sometimes "changed" events are emitted on files that were actually deleted,
             //so determine file existance and act accordingly
-            if (await util.fileExists(change.pathAbsolute)) {
-                await program.addOrReplaceFile({
-                    src: change.pathAbsolute,
-                    dest: rokuDeploy.getDestPath(change.pathAbsolute, options.files, rootDir)
-                });
+            if (await util.pathExists(change.pathAbsolute)) {
```

Comment:
I realize: why do you test if the file physically exists instead of letting the file resolver(s) try to find it? Otherwise what's the point to the list of resolvers?

Reply:
The file resolver concept was created to support the language server's TextDocument design. Generally you'll only have two resolvers: the language server, and then the actual file system. We read the language server first, and if not there, go to the file system.

Since our language server has a separate process to handle not-saved transient files (I call them standalone files), we can safely assume that all other files are physically located on disk, regardless of whether they are currently opened in the editor or not, so this check works just fine.

Unless I'm missing something, I don't see any need to modify the file resolvers to support doesFileExist logic since it doesn't seem to provide any benefit. Do you?

Reply:
Ok makes sense.